addons/mail/static/src/components/attachment_viewer/attachment_viewer.js

```diff
@@ -31,6 +31,10 @@ class AttachmentViewer extends Component {
                 attachmentViewer: attachmentViewer ? attachmentViewer.__state : undefined,
             };
         });
+        /**
+         * Used to ensure that the ref is always up to date, which seems to be needed if the element
+         * has a t-key, which was added to force the rendering of a new element when the src of the image changes.
```

Comment:
add the reason of the src change (-> show loading instead of previous image while it is loading)